satellite/satellitedb/metainfo_buckets.go

```diff
@@ -88,11 +100,23 @@ func (buckets *buckets) List(ctx context.Context, projectID uuid.UUID, opts meta
 			limit, 0,
 		)
 	case storj.Backward:
-		dbxBuckets, err = buckets.db.Limited_Bucket_By_ProjectId_And_Name_LessOrEqual_OrderBy_Asc_Name(ctx,
-			dbx.Bucket_ProjectId(projectID[:]),
-			dbx.Bucket_Name([]byte(opts.Cursor)),
-			limit, 0,
-		)
+		// TODO most probably needs optimization
+		if opts.Cursor == "" {
+			dbxBuckets, err = buckets.db.Limited_Bucket_By_ProjectId_And_Name_GreaterOrEqual_OrderBy_Desc_Name(ctx,
+				dbx.Bucket_ProjectId(projectID[:]),
+				dbx.Bucket_Name([]byte(opts.Cursor)),
+				limit, 0,
+			)
+		} else {
+			dbxBuckets, err = buckets.db.Limited_Bucket_By_ProjectId_And_Name_LessOrEqual_OrderBy_Desc_Name(ctx,
+				dbx.Bucket_ProjectId(projectID[:]),
+				dbx.Bucket_Name([]byte(opts.Cursor)),
+				limit, 0,
+			)
+		}
+		sort.Slice(dbxBuckets, func(i, j int) bool {
```

Comment:
reverse the order instead of using sort

Reply:
With just reverse order when we have set `"a", "aa", "b", "bb", "c"` and we have cursor `c`, direction `Backward` and limit `1` result will be `a` not `c`

Reply:
I mean just replacing the sort.Slice